parser/parser.go

```diff
@@ -45,12 +47,27 @@ type Test struct {
 	Time int // in milliseconds
 }
 
+// Benchmark contains the results of a single benchmark.
+type Benchmark struct {
+	Name     string
+	Duration time.Duration
+	// number of B/op
+	Bytes int
+	// number of allocs/op
+	Allocs int
+	// number of times this benchmark has been seen (for averaging).
+	Count  int
+	Output []string
+}
+
 var (
 	regexStatus   = regexp.MustCompile(`--- (PASS|FAIL|SKIP): (.+) \((\d+\.\d+)(?: seconds|s)\)`)
 	regexCoverage = regexp.MustCompile(`^coverage:\s+(\d+\.\d+)%\s+of\s+statements(?:\sin\s.+)?$`)
 	regexResult   = regexp.MustCompile(`^(ok|FAIL)\s+([^ ]+)\s+(?:(\d+\.\d+)s|\(cached\)|(\[\w+ failed]))(?:\s+coverage:\s+(\d+\.\d+)%\sof\sstatements(?:\sin\s.+)?)?$`)
-	regexOutput   = regexp.MustCompile(`(    )*\t(.*)`)
-	regexSummary  = regexp.MustCompile(`^(PASS|FAIL|SKIP)$`)
+	// regexBenchmark captures 3-5 groups: benchmark name, number of times ran, ns/op (with or without decimal), B/op (optional), and allocs/op (optional).
+	regexBenchmark = regexp.MustCompile(`^(Benchmark\w+)-\d\s+(\d+)\s+((\d+|\d+\.\d+)\sns/op)(\s+\d+\sB/op)?(\s+\d+\sallocs/op)?`)
```

Comment:
We can adjust the regex pattern a bit so that we don't have to trim any suffixes from the captured groups.

Something like this could work:
```go
`^(Benchmark[^ ]+)\s+(\d+)(?:\s+(\d+|\d+\.\d+)\s[a-zA-Z]+/op)(?:\s+(\d+|\d+\.\d+)\s[a-zA-Z]+/op)?(?:\s+(\d+|\d+\.\d+)\s[a-zA-Z]+/op)?$`
```
This will result in 5 capture groups that can all be directly used without much processing (benchmark name, and the numbers representing iteration count, ns/op, B/op, allocs/op).